frontend/src/pages/mainPage.js

```diff
@@ -1,32 +1,57 @@
-import React, {useState, useEffect} from "react";
+import React, {useState, useEffect, createContext, useContext} from "react";
 import { Link } from "react-router-dom";
 import axios from 'axios';
-
-import News from "../components/mainPage/News.js"
-
-/*const news =[
-    {
-        img: "/img/team.jpg",
-        title: "Why did the programmer quit his job? Because he didn't get arrays.",
-        date: "02.04.2006"
-    },
-    {
-        img: "/img/team.jpg",
-        title: "Why do Java developers wear glasses? Because they can't C#.",
-        date: "11.12.2005"
-    },
-    {
-        img: "/img/team.jpg",
-        title: "There are only 10 types of people in the world: those who understand binary and those who don't.",
-        date: "14.02.2006"
-    }
-
-
-];*/
-
-
-function MainPage({news}) {
-    const [loginStatus,setLoginStatus]=useState(true)
+import apiEndpoints from "../apiEndpoints";
+
+import News from "../components/mainPage/news.js"
+import SportBtn from "../components/mainPage/sportBtn"
+import Stream from "../components/mainPage/stream.js"
+
+function MainPage() {
+    const [loginStatus,setLoginStatus]=useState(false)
+
+    const [news, setNews] = useState([]);
+    const [sports, setSport] = useState([]);
+    const [games, setGames] = useState([]);
+
+    useEffect(() => {
+        axios.get(`${apiEndpoints.url}${apiEndpoints.news.getRecent}`)
+            .then(res => {
+                const returnedNews = res.data;
+                setNews(returnedNews);
+            })
+            .catch(error => {
+                console.error('There was an error getting news:', error);
```

Comment:
Think about implementing some sort of a pop-up (any kind will do) to display those errors to the end user.

Regular customer does not know what console is let alone how to open it

Reply:
Done